linker-interface/shared/src/main/scala/org/scalajs/linker/interface/ReportToLinkerOutputAdapter.scala

```diff
@@ -0,0 +1,210 @@
+/*
+ * Scala.js (https://www.scala-js.org/)
+ *
+ * Copyright EPFL.
+ *
+ * Licensed under Apache License 2.0
+ * (https://www.apache.org/licenses/LICENSE-2.0).
+ *
+ * See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ */
+
+package org.scalajs.linker.interface
+
+import scala.concurrent._
+
+import java.net.URI
+import java.nio.ByteBuffer
+import java.nio.charset.StandardCharsets.UTF_8
+
+import org.scalajs.logging.Logger
+
+import org.scalajs.linker.interface.unstable.{OutputDirectoryImpl, OutputFileImpl}
+
+/** Backwards compatibility implementation for pre 1.3.0 link method.
+ *
+ *  The major interface change in 1.3.0 is that the linker (and not the
+ *  caller) determines the set of files to be written. As a consequence, the
+ *  post 1.3.0 API does not offer as much control over cross-file references
+ *  (i.e. source map links): it is based on patterns rather than simply
+ *  asking the caller to verbatim provide the URI to reference in each file.
+ *
+ *  To provide a backwards compatible interface, we do the following post-run
+ *  processing:
+ *
+ *  - Match and copy the produced set of files (in the OutputDirectory) to
+ *    the files provided by the caller (in LinkerOutput).
+ *  - Replace the pattern generated cross-file references with the ones
+ *    provided by the caller. This is necessary as a post-processing step,
+ *    because of the reduced flexibility of the 1.3.0 API: we cannot express
+ *    all legacy requests in the new API.
+  */
+@deprecated("Part of legacy API.", "1.3.0")
+object ReportToLinkerOutputAdapter {
+  final class UnsupportedLinkerOutputException private[ReportToLinkerOutputAdapter] (
+      message: String) extends IllegalArgumentException(message)
+
+  def convert(report: Report, outputDirectory: OutputDirectory,
+      legacyOutput: LinkerOutput)(
+      implicit ec: ExecutionContext): Future[Unit] = {
+    retrieveOutputFiles(report, outputDirectory).flatMap {
+      case (jsFileContent, optSourceMapContent) =>
+
```

Comment:
And it's not a big deal, but since there is something else to change anyway: this blank line is not supposed to be there, ideally.